lib/matplotlib/widgets.py

```diff
@@ -3443,6 +3457,66 @@ def __init__(self, ax, onselect, useblit=False,
         self.grab_range = grab_range
 
         self.set_visible(True)
+        self._draw_box = draw_box
+        self._box = None
+
+    def _get_bbox(self):
+        return self._selection_artist.get_bbox()
+
+    def _add_box(self):
+        self._box = RectangleSelector(self.ax,
+                                      onselect=self._scale_polygon,
+                                      useblit=self.useblit,
+                                      interactive=True)
+        self._update_box()
+        # Set state that prevents the RectangleSelector from being created
+        # by the user
+        self._box._allow_creation = False
+        self._box._selection_completed = True
+        self._draw_polygon()
+
+    def _remove_box(self):
+        if self._box is not None:
+            self._box.set_visible(False)
+            del self._box
```

Comment:
Something I am not sure: is it necessary to delete before setting `self._box` to `None`?

Reply:
Maybe not, I will check...

Reply:
No, it wasn't necessary, thanks for catching!